### ISSUE TITLE
Fix logger issue printing 'NoneType: None' in error logs

### ISSUE THREAD

**Issue by yanazaeva** [wazuh/wazuh#12790]
|Wazuh version|Component|
|---|---|
| master | Wazuh cluster |

## Description
It seems that whenever the debug mode is enabled, the error messages are not being printed properly:
```
2022/03/21 08:46:49 INFO: [Worker] [Main] Connection from ('172.20.0.4', 57858)
2022/03/21 08:46:49 ERROR: [Worker] [Main] #### Test error
NoneType: None
2022/03/21 08:46:49 ERROR: [Worker] [Main] Error during handshake with incoming connection.
NoneType: None
2022/03/21 08:46:49 ERROR: [Worker] [Main] #### Another test error
NoneType: None
```

These two errors were obtained by modifying the `connection_lost` method from the `AbstractServerHandler` class:
```PYTHON3
    def connection_lost(self, exc):
        """Define process of closing connection with the server.

        Remove client from global clients dictionary and log the exception if any.

        Parameters
        ----------
        exc : Exception
            In case the connection was lost due to an exception, it will be contained in this variable.
        """
        if self.name:
            if exc is None:
                self.logger.debug(f"Disconnected {self.name}.")
            else:
                self.logger.error(f"Error during connection with '{self.name}': {exc}.\n"
                                  f"{''.join(traceback.format_tb(exc.__traceback__))}")

            if self.name in self.server.clients:
                del self.server.clients[self.name]
        else:
            if exc is not None:
                self.logger.error(f"Error during handshake with incoming connection: {exc}", exc_info=True)
            else:
                self.logger.error(f'#### Test error')
                self.logger.error("Error during handshake with incoming connection.")

        self.logger.error("#### Another test error")
```
It seems that his problem remains in the following method:
https://github.com/wazuh/wazuh/blob/acc66b0248167dd7b30a5ec1fded87c2cc750629/framework/wazuh/core/wlogging.py#L86-L123
We should investigate the reason behind it and find the best way of handling the error. 

### wazuh/wazuh
- Unit tests without failures. Updated and/or expanded if there are new functions/methods/outputs:
  - [ ] Cluster (`framework/wazuh/core/cluster/tests/` & `framework/wazuh/core/cluster/dapi/tests/`)
  - [ ] Core (`framework/wazuh/core/tests/`)
  - [ ] SDK (`framework/wazuh/tests/`)
  - [ ] RBAC (`framework/wazuh/rbac/tests/`)
  - [ ] API (`api/api/tests/`)
- API tavern integration tests without failures. Updated and/or expanded if needed (`api/test/integration/`):
  - [ ] Affected tests 
  - [ ] Affected RBAC (black and white) tests
- [ ] Review integration test mapping using the script (`api/test/integration/mapping/integration_test_api_endpoints.json`)
- [ ] Review of spec.yaml examples and schemas (`api/api/spec/spec.yaml`)
- [ ] Review exceptions remediation when any endpoint path changes or is removed (`framework/wazuh/core/exception.py`)
- [ ] Changelog (`CHANGELOG.md`)

### wazuh/wazuh-documentation
- [ ] Migration from 3.X for changed endpoints (`source/user-manual/api/equivalence.rst`)
- [ ] Update RBAC reference with new/modified actions/resources/relationships (`source/user-manual/api/rbac/reference.rst`)

**Comment by yanazaeva** [comment]
## Issue update

Here, we will study how removing these two lines will affect other parts of the code, as directly removing it might not be the best solution:
https://github.com/wazuh/wazuh/blob/acc66b0248167dd7b30a5ec1fded87c2cc750629/framework/wazuh/core/wlogging.py#L114-L115
In the comment below, It was specified that only when there is no error the `NoneType: None` message is being displayed. However, this option should be available by default if we are in `debug2`. 

For example, in `debug2`, if we obtain an error in the following lines:
https://github.com/wazuh/wazuh/blob/875f47ee8de1bad0cff5db4019a28f6967213660/framework/scripts/wazuh-clusterd.py#L206-L208
we expect to have the following output in the logs:
```
2022/03/23 08:54:21 ERROR: [Cluster] [Main] Directory '/tmp' needs read, write & execution permission for 'wazuh' user
Traceback (most recent call last):
  File "/var/ossec/framework/scripts/wazuh-clusterd.py", line 203, in <module>
    raise MemoryError()
MemoryError
```
So the condition we wanted to delete before has to remain. However, a proper solution would be to check for all the `logger.error` messages and make sure that if they are not being displayed due to an error, they will contain an `exc_info` field, so the `NoneType: None` message is not shown. 

Also, for the lines below, it was noticed that the traceback will always be `NoneType: None`, as the `[Errno 111] Connection refused` error seems not to have any traceback. Due to this, `exc_info` will be set to `False` in both lines:
https://github.com/wazuh/wazuh/blob/875f47ee8de1bad0cff5db4019a28f6967213660/framework/wazuh/core/cluster/server.py#L186-L190
- Output log:
```
2022/03/23 08:54:06 INFO: [Worker] [Main] Connection from ('172.19.0.5', 56268)
2022/03/23 08:54:06 ERROR: [Worker] [Main] Error during handshake with incoming connection: [Errno 111] Connection refused
NoneType: None
```

Below I will leave some of the changes explained:
<details>
<summary>Expand changes in `wazuh-clusterd.py`</summary>

- Before the change:
```
root@bc8ddedce15e:/# /var/ossec/bin/wazuh-clusterd -f
2022/03/23 21:25:02 ERROR: [Cluster] [Main] Cluster is already running.
NoneType: None
```
- After the change
```
root@bc8ddedce15e:/# /var/ossec/bin/wazuh-clusterd -f
2022/03/23 21:32:14 ERROR: [Cluster] [Main] Cluster is already running.
root@bc8ddedce15e:/# 
```

</details>

<details>
<summary>Expand changes in `common.py`</summary>

- Before the change:
```
2022/03/23 13:08:28 ERROR: [Worker 3d1974982777] [Main] The task was canceled due to the following error on the remote node: error
NoneType: None

```
- After the change
```
2022/03/23 13:07:06 ERROR: [Worker 99f99bd89afd] [Main] The task was canceled due to the following error on the remote node: error
```

</details>

In a nutshell, we can distinguish between three two cases:
- When we store an Exception in a variable or use a message such as `Error was found`: for these scenarios, the `exc_info` has to be set to `False` as the unwanted message will be printed. 
- The exception is raised, so the traceback can be properly found.
